### ISSUE TITLE
Zooming-in > 400% consumes > 4 GB of memory, may lead to whole session crash

### ISSUE THREAD

**Issue by Amr-Ibra** [xournalpp/xournalpp#5161]
### Operating System

Linux

### (Linux only) Distribution

Debian testing

### (Linux only) Desktop Environment

GNOME

### (Linux Only) Display Server

Wayland

### Installation Method

Debian package

### Xournal++ Version

1.2.1

### libgtk Version

3.24.38

### Bug Description

Zooming-in > 400% consumes > 4 GB of memory, may lead to session crash

### Expected Behaviour

* Zooming-in indefinitely does not bring any value
* Zooming-in should have a maximum limit
* Memory usage should not grow out of proportion

### Steps to Reproduce

1. Open a PDF file in Xournal++ (example file: [Get_Started_With_Smallpdf.pdf](https://github.com/xournalpp/xournalpp/files/12665739/Get_Started_With_Smallpdf.pdf))
2. Zoom-in indefinitely
3. Watch the memory usage of Xournal++ grow out of proportion
4. Whole session may crash if runs out of memory

**Comment by KlausBlum** [comment]
On my (Windows 10) system, zooming is limited to 700%. 
Memory consumption then is 3923,7 MB using your example file (but no crash).

**Comment by Amr-Ibra** [comment]
> On my (Windows 10) system, zooming is limited to 700%.
> Memory consumption then is 3923,7 MB using your example file (but no crash).

In my humble opinion, zooming this far, and consuming this amount of memory, both should never happen. You'd probably not notice any zoom difference beyond a specific zoom level. 4 GB memory is just too much. And crashing depends on the total system memory (available free memory in the system for everything else).

**Comment by Amr-Ibra** [comment]
> On my (Windows 10) system, zooming is limited to 700%. Memory consumption then is 3923,7 MB using your example file (but no crash).

Did you try to zoom back to 100% and watch the memory usage? Has the memory been released (decreased)?

**Comment by Amr-Ibra** [comment]
> Memory consumption then is 3923,7 MB using your example file (but no crash).

Did you try to zoom-out back to 100% and watch the memory usage? Has the memory been released (decreased)?

**Comment by bhennion** [comment]
This is a known issue that will take a lot of time to fix. It requires an in depth refactoring of the view system. Some progress has been made towards that but a fix is not yet around the corner. Don't hold your breaths. It will take a lot of time.

**Comment by Amr-Ibra** [comment]
> Yes, it decreases, but only a small amount. Still about 3100 MB with zoom below 100%.

Still too much.

**Comment by toprak** [comment]
having same issue on arch linux with x11 i3wm

**Comment by j-lakeman** [comment]
> What I found most interesting is that there seems to be a zoom limit in Windows, but not in Linux.

It would be quite helpful to have this on Linux, too, IMHO. To prevent oneself from accidentally zooming in too far and thus some sort of workaround to limit the memory usage.

**Comment by j-lakeman** [comment]
True, but i mean implementing an option to manually set the max zoom to say 300% in order to avoid heavy memory usage.

**Comment by benniju** [comment]
I was able to reduce the memory usage from 5 GB when zoomed in 700% to "just" 800 MB by setting the preloading of the previous and next page to 0. I am not noticing any loge wait times because of this, probably because its reading from an SSD.


I think this issue might be related to my other issue #5389.
